modules/core/arrow-test/src/main/kotlin/arrow/test/laws/BimonadLaws.kt

```diff
@@ -14,13 +14,13 @@ object BimonadLaws {
     BF: Bimonad<F>,
     M: Monad<F>,
     CM: Comonad<F>,
-    f: (Int) -> Kind<F, Int>,
+    G: Gen<Kind<F, Int>>,
     EQ1: Eq<Kind<F, Int>>,
     EQ2: Eq<Kind<F, Kind<F, Int>>>,
     EQ3: Eq<Int>
```

Comment:
All of these can be collapsed into simply `EqK<F>` and `Eq<Int>` is not needed as a parameter at all

Reply:
ok, i have made the changes to BimonadLaws now. Now there is an issue that i have run into before (and was one of my open questions):

`NonEmptyListTest` now uses lists of size > 1. This leads to failures of the test ComonadLaws.cobinding. Honestly i do not understand the intention of the test and what needs to be changed to fix it.

Reply:
Honestly I know far too little about comonads to look into this. Since this pr touches almost every datatype this is going to get really interesting :) I think @Cotel and @pakoito did most of the comonad code, so they might be able to help here.

Reply:
So after talking to paco on slack, I think I know whats wrong. I won't go into much detail, the test assumption is plain wrong. Can you comment out the `cobinding` test with a note to add an issue? I'll open it and fill in all relevant information.